File: VisualStudioCodeEditor/01-Editors/13-RefactorYourCode/Start (1)/RefactorMyVB/Book.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.IO

Namespace RefactorMyVB

  Friend Class Book
	Public Property Title() As String
	Public Property Price() As Decimal

	' rename parameter  (overwriteExisting).  It has a typo
	' reorder parameters
	Public Sub SaveBook(ByVal fileLocation As String, ByVal overwriteExising As Boolean, ByVal timeStamp As Date)
	  Dim fs As FileStream
	  If overwriteExising Then
		fs = New FileStream(fileLocation, FileMode.Create)
	  Else
		fs = New FileStream(fileLocation, FileMode.Open)

	  End If


	End Sub
  End Class
End Namespace

File: VisualStudioCodeEditor/01-Editors/13-RefactorYourCode/Start (1)/RefactorMyCSharp/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace RefactorMyCSharp
{
  class Book
  {
    public string Title { get; set; }
    public decimal Price { get; set; }

    // rename parameter  (overwriteExisting).  It has a typo
    // reorder parameters
    public void SaveBook(string fileLocation, bool overwriteExising, DateTime timeStamp)
    {
      FileStream fs;
      if (overwriteExising)
      {
        fs = new FileStream(fileLocation, FileMode.Create);
      }
      else
      {
        fs = new FileStream(fileLocation, FileMode.Open);

      }


    }
  }
}
